fdb-record-layer-lucene/src/main/java/com/apple/foundationdb/record/lucene/codec/LuceneOptimizedWrappedDirectory.java

```diff
@@ -0,0 +1,139 @@
+/*
+ * WrappedDirectory.java
+ *
+ * This source file is part of the FoundationDB open source project
+ *
+ * Copyright 2015-2021 Apple Inc. and the FoundationDB project authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.apple.foundationdb.record.lucene.codec;
+
+import com.apple.foundationdb.record.lucene.directory.FDBDirectory;
+import org.apache.lucene.store.BufferedChecksumIndexInput;
+import org.apache.lucene.store.ChecksumIndexInput;
+import org.apache.lucene.store.Directory;
+import org.apache.lucene.store.FilterDirectory;
+import org.apache.lucene.store.IOContext;
+import org.apache.lucene.store.IndexInput;
+import org.apache.lucene.store.IndexOutput;
+import org.apache.lucene.store.Lock;
+
+import java.io.IOException;
+import java.util.Collection;
+
+import static com.apple.foundationdb.record.lucene.codec.LuceneOptimizedCompoundFormat.DATA_EXTENSION;
+import static com.apple.foundationdb.record.lucene.codec.LuceneOptimizedCompoundFormat.ENTRIES_EXTENSION;
+import static org.apache.lucene.codecs.lucene70.Lucene70SegmentInfoFormat.SI_EXTENSION;
+
+public class LuceneOptimizedWrappedDirectory extends Directory {
+    private final FDBDirectory fdbDirectory;
+    private final Directory wrappedDirectory;
+
+    public static boolean isSegmentInfo(String name) {
+        return name.endsWith(SI_EXTENSION);
+    }
+
+    public static boolean isCompoundFile(String name) {
+        return name.endsWith(DATA_EXTENSION);
+    }
+
+    public static boolean isEntriesFile(String name) {
+        return name.endsWith(ENTRIES_EXTENSION);
+    }
+
+    public static String convertToDataFile(String name) {
+        if (isSegmentInfo(name)) {
+            return name.substring(0, name.length() - 2) + DATA_EXTENSION;
+        } else if (isEntriesFile(name)) {
+            return name.substring(0, name.length() - 3) + DATA_EXTENSION;
+        } else {
+            return name;
+        }
+
+    }
+
+    public LuceneOptimizedWrappedDirectory(Directory directory) {
+        this.wrappedDirectory = directory;
+        this.fdbDirectory = (FDBDirectory)FilterDirectory.unwrap(directory);
+    }
+
+    @Override
```

Comment:
Is there a reason we're specifically calling these out as unsupported?

Reply:
The Wrapped Directory was for input and output in a specific case and I did not want to think this is a general directory implementation.